api.c

```diff
@@ -21,7 +21,11 @@ int api_bindlisten(const char *api_socket)
     }
     memset(&addr, 0, sizeof(addr));
     addr.sun_family = AF_UNIX;
-    strncpy(addr.sun_path, api_socket, sizeof(addr.sun_path) - 1);
+    if (strlen (api_socket) >= sizeof(addr.sun_path)) {
+        fprintf(stderr, "the specified API socket path is too long (> %d)\n", sizeof(addr.sun_path));
+        return -1;
+    }
+    strcpy(addr.sun_path, api_socket);
```

Comment:
I think we should use strncpy everytime to avoid potential lint false-positive

Reply:
also make sure to terminate with NUL

Reply:
moved back to strncpy.

The `addr.sun_path` string is NUL terminated since we have `memset(&addr, 0, sizeof(addr));`